internal/restconfig/restconfig.go

```diff
@@ -256,121 +320,109 @@ func (rcp *Producer) RunOnSelectedPrefixedContext(prefix string, function PerCon
+		return true, runInCluster(function, status)
+	}
 
-	if usage == "" {
-		usage = "comma-separated list of kubeconfig contexts to use, can be specified multiple times.\n" +
-			"If none specified, all contexts referenced by the kubeconfig are used"
+	if rcp.defaultClientConfig.overrides.CurrentContext != "" {
+		// The user has explicitly chosen a context, use that only
+		return true, rcp.RunOnSelectedContext(function, status)
 	}
 
-	cmd.PersistentFlags().StringSliceVar(&rcp.kubeContexts, "kubecontexts", nil, usage)
-}
+	if rcp.defaultClientConfig == nil {
+		return false, nil
+	}
 
-// AddInClusterConfigFlag adds a flag enabling in-cluster configurations for processes running in pods.
-func (rcp *Producer) AddInClusterConfigFlag(cmd *cobra.Command) {
-	cmd.PersistentFlags().BoolVar(&rcp.inCluster, "in-cluster", false, "use the in-cluster configuration to connect to Kubernetes")
-}
+	clientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		rcp.defaultClientConfig.loadingRules, rcp.defaultClientConfig.overrides)
 
-func (rcp *Producer) PopulateTestFramework() {
-	framework.TestContext.KubeContexts = rcp.kubeContexts
-	if rcp.kubeConfig != "" {
-		framework.TestContext.KubeConfig = rcp.kubeConfig
+	rawConfig, err := clientConfig.RawConfig()
+	if err != nil {
+		return false, status.Error(err, "error retrieving the raw kubeconfig setup")
 	}
-}
 
-func (rcp *Producer) MustGetForClusters() []RestConfig {
-	configs, err := rcp.getRestConfigs()
-	exit.OnErrorWithMessage(err, "Error getting REST Config for cluster")
+	contextErrors := []error{}
+	processedContexts := 0
 
-	return configs
-}
+	if len(rcp.contexts) > 0 {
+		// Loop over explicitly-chosen contexts
+		for _, contextName := range rcp.contexts {
+			chosenContext, ok := rawConfig.Contexts[contextName]
+			if !ok {
+				status.Failure("error retrieving context %s", contextName)
 
-func (rcp *Producer) CountRequestedClusters() int {
-	if len(rcp.kubeContexts) > 0 {
-		// Count unique contexts
-		contexts := stringset.New()
-		for i := range rcp.kubeContexts {
-			contexts.Add(rcp.kubeContexts[i])
-		}
-
-		return contexts.Size()
-	}
-	// Current context or rcp.kubeContext
-	return 1
-}
+				continue
+			}
 
-func (rcp *Producer) forCluster() (RestConfig, error) {
-	var restConfig RestConfig
+			processedContexts++
 
-	restConfigs, err := rcp.getRestConfigs()
-	if err != nil {
-		return restConfig, err
-	}
+			contextErrors = append(contextErrors, rcp.overrideContextAndRun(chosenContext.Cluster, contextName, function, status))
+		}
+	} else {
+		// Loop over all accessible contexts
+		for contextName, context := range rawConfig.Contexts {
+			processedContexts++
 
-	if len(restConfigs) > 0 {
-		return restConfigs[0], nil
+			contextErrors = append(contextErrors, rcp.overrideContextAndRun(context.Cluster, contextName, function, status))
+		}
 	}
 
-	return restConfig, errors.New("error getting restconfig")
+	return processedContexts > 0, k8serrors.NewAggregate(contextErrors)
```

Comment:
So all the call sites want to `failIfNoContext` - I don't see a case where we wouldn't - so I suggest we just handle it here instead of returning a boolean and requiring callers to do it.